Modules/posixmodule.c

```diff
@@ -5533,12 +5547,13 @@ Execute the program specified by path in a new process.
 static PyObject *
 os_posix_spawn_impl(PyObject *module, path_t *path, PyObject *argv,
                     PyObject *env, PyObject *file_actions,
-                    PyObject *setpgroup, int resetids, PyObject *setsigmask,
-                    PyObject *setsigdef, PyObject *scheduler)
-/*[clinic end generated code: output=45dfa4c515d09f2c input=2891c2f1d457e39b]*/
+                    PyObject *setpgroup, int resetids, int setid,
+                    PyObject *setsigmask, PyObject *setsigdef,
+                    PyObject *scheduler)
+/*[clinic end generated code: output=f681e0e4594b8038 input=bfb7dcbaca90eb8a]*/
 {
     return py_posix_spawn(0, module, path, argv, env, file_actions,
-                          setpgroup, resetids, setsigmask, setsigdef,
+                          setpgroup, resetids, setsid, setsigmask, setsigdef,
```

Comment:
`os_posix_spawn_impl` accepts `setid` instead of `setsid`. Your code currently passes a pointer to `setsid()` standard function to `py_posix_spawn()` instead of `setid` argument. This is probably why you added `intptr_t` cast initially. It's better to rename the `setsid` parameter so that the standard function is not shadowed (e.g. `call_setsid`).